File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
def dfs(numbers, target, curr, idx):
    answer = 0
    
    if idx == len(numbers):
        if curr == target:
            answer += 1
        return answer
    
    answer += dfs(numbers, target, curr + numbers[idx], idx+1)
    answer += dfs(numbers, target, curr - numbers[idx], idx+1)
    return answer

def solution(numbers, target):
    answer = dfs(numbers, target, 0, 0)
    return answer

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        int answer = dfs(numbers, target, 0, 0);
        return answer;
    }
    
    int dfs(int[] numbers, int target, int curr, int idx) {
        int answer = 0;
        
        if (idx == numbers.length) {
            if (curr == target) {
                answer++;
            }
        } else {
            answer += dfs(numbers, target, curr + numbers[idx], idx+1);
            answer += dfs(numbers, target, curr - numbers[idx], idx+1);
        }
        
        return answer;
    }
}